File: baekjoon/Basic/BinaryNumber.cpp
/* 3460 이진수 
십진수 이진수로 바꾸기.
 testcase별 출력에 개행문자 포함하는 것 잊지말기.
*/
#include <iostream>
#include <vector>
using namespace std;

vector<int> binary_num;

void dec_to_bin(int num){
    int divided=num;
    while(divided!=1){
        binary_num.push_back(divided%2);
        divided=divided/2;
    }
    binary_num.push_back(1);
}

int main(){
    int testcase;

    cin>>testcase;
    for(int i=0; i<testcase; i++){
        binary_num.clear();
        int n;
        cin>>n;
        dec_to_bin(n);
        for(int i=0; i<binary_num.size(); i++){
            if(binary_num[i]==1) {
                cout<<i;
                if(i==binary_num.size()-1) cout<<'\n';
                else cout<<' ';
            }
        }
    }
   
    return 0;
}

File: baekjoon/Basic/BinaryNumber.java
package baekjoon.Basic;
import java.io.*;
import java.util.*;

public class BinaryNumber {
    static ArrayList<Integer> binary_num=new ArrayList<Integer>();
    static void dec_to_bin(int num){
        int divided=num;
        while(divided!=1){
            binary_num.add(divided%2);
            divided=divided/2;
        }
        binary_num.add(1);
    }
    public static void main(String[] args) throws IOException{
        BufferedReader br=new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st=new StringTokenizer(br.readLine());
        int testCase=Integer.parseInt(st.nextToken());
        for(int i=0; i<testCase; i++){
            binary_num.clear();
            st=new StringTokenizer(br.readLine());
            int N=Integer.parseInt(st.nextToken());
            dec_to_bin(N);
            for(int j=0; j<binary_num.size(); j++){
                if(binary_num.get(j)==1){
                    System.out.print(j);
                    if(j==binary_num.size()-1) System.out.print("\n");
                    else System.out.print(" ");
                }
            }
            
        }
    }
}
